pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -47,29 +47,19 @@ const (
 func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, proxy *model.Proxy, push *model.PushContext) ([]*v2.Cluster, error) {
 	clusters := make([]*v2.Cluster, 0)
 
-	recomputeOutboundClusters := true
-	if configgen.OutboundClusters != nil {
-		clusters = append(clusters, configgen.OutboundClusters...)
-		recomputeOutboundClusters = false
-	}
-
-	if recomputeOutboundClusters {
-		clusters = append(clusters, configgen.buildOutboundClusters(env, push)...)
-	}
-
-	// If its a multi-cluster gateway, do not enable mTLS
-	// for clusters, as the gateway is already receiving mTLS
-	// traffic from downstream
-	if proxy.GetGatewayMode() == model.MulticlusterGateway {
-		// TODO: cache this output just like the output
-		sniClusters := make([]*v2.Cluster, len(clusters))
-		for _, c := range clusters {
-			// copy each cluster, unset the tls context
-			newC := *c
-			newC.TlsContext = nil
-			sniClusters = append(sniClusters, &newC)
+	if proxy.HasSnowFlakes() {
+		// No caching for this proxy.
+		clusters = append(clusters, configgen.buildOutboundClusters(env, proxy, push)...)
```

Comment:
This will rebuild all the clusters. With the previous flag, no need to rebuild everything.

Reply:
? It will rebuild the clusters only for the mcgateway. The memory requirements are the same in both cases isn't it?

Reply:
I agree. But the previous approach is much simpler with a bit less computation (which may not be critical).